notification/app/notification/services/guardian/GuardianNotificationSender.scala

```diff
@@ -112,7 +112,7 @@ class GuardianNotificationSender(
     }
 
     shard(countWithDefault).map { shard =>
-      val shardedNotification = ShardedNotification(notification, shard, Some(notificationReceivedTime))
+      val shardedNotification = ShardedNotification(notification, shard, Some(NotificationMetadata(notificationReceivedTime, registrationCount)))
```

Comment:
As an aside, the notification API does not query the database at real time to get this `registrationCount`, but retrieves this value from a S3 bucket which are updated by the `topicCounter` lambda regularly.  So strictly speaking, the `registrationCount` value may differ from the total number of devices the sender worker actually attempt to send, but I think it is more than good enough to be used to group notifications based on audience size in this context.

Reply:
Thanks for the extra context @waisingyiu. As you mentioned, I think this will be accurate enough for our purposes, but it's good to understand how this actually works!